app/src/main/java/com/android/voyageur/model/trip/TripsViewModel.kt

```diff
@@ -181,6 +191,50 @@ open class TripsViewModel(
     }
   }
 
+  open fun getPhotosForSelectedTrip(): List<String> {
+    return selectedTrip.value?.photos ?: emptyList()
+  }
+
+  open fun addPhotoToTrip(photo: String) {
+    if (selectedTrip.value != null) {
+      val trip = selectedTrip.value!!
+      val updatedTrip = trip.copy(photos = trip.photos + photo)
+      updateTrip(
+          updatedTrip,
+          onSuccess = {
+            /*
+                This is a trick to force a recompose, because the reference wouldn't
+                change and update the UI.
+            */
+            selectTrip(Trip())
+            selectTrip(updatedTrip)
+          },
+          onFailure = { error ->
+            Log.e("PhotosScreen", "Error adding photo: ${error.message}", error)
+          })
+    }
+  }
+
+  open fun removePhotoFromTrip(photo: String) {
+    if (selectedTrip.value != null) {
```

Comment:
same suggestions as above

Reply:
I made the same fixes as in addPhotoToTrip